test/internals/reply.test.js

```diff
@@ -729,6 +729,30 @@ test('non-string with content type application/json SHOULD be serialized as json
   })
 })
 
+test('non-string with custom json\'s content-type SHOULD be serialized as json', t => {
+  t.plan(4)
+
+  const fastify = require('../..')()
+
+  fastify.get('/', function (req, reply) {
+    reply.type('application/json; version=2; ').send({ key: 'hello world!' })
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+    fastify.server.unref()
```

Comment:
I know it is used in other tests, but could you use the `t.teaddown` function instead of `server.unref`?

Reply:
PTAL, but I want to know what is the difference between `t.teardown` & `server.unref`, more semantic?

Reply:
`t.teardown` run when the test end.
`server.unref` just didn't block the program end when no code need to run inside the event loop. It is an unclean close.